PosteAPI/serializers.py

```diff
@@ -110,13 +111,70 @@ class Meta:
         fields = "__all__"
 
 
+class PostCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Post
+        fields = "__all__"
+        extra_kwargs = {
+            "title": {"required": True},
+            "url": {"required": True},
+        }
+
+    def create(self, validated_data):
+        post = Post(
+            title=validated_data["title"],
+            description=validated_data["description"],
+            url=validated_data["url"],
+            creator=validated_data["creator"],
+            folder=validated_data["folder"]
+        )
+        return post
+
+    def validate_url(self, value):
+        url_form_field = URLField()
+        try:
+            url = url_form_field.clean(value)
+        except ValidationError:
+            raise serializers.ValidationError("invalid url")
+        return url
+
+    def validate_user(self, value):
+        try:
+            return User.objects.get(pk=value.pk)
```

Comment:
We would need to send the id since that would be the only unique identifier for Folder. The folder serialize already sends the id to the client will have it.